### PR TITLE
Config policy testing docs

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description
* Added page on writing an running tests for config policies, and added to the sidebar
* Added references to testing to the config policy overview page
* Changed the name of the how-to on writing policies to better reflect what's on the page

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/test-config-policies.adoc [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+7/7 tests passed (0.001s)
+----
+
+. To build more confidence, best practice is to create a top level test that will use the entire policy bundle, similar to an integration or end-to-end test.
+. Create a new test file: ./config-policies/policy_test.yaml`
+. Copy the following into your `policy_test.yaml` file:
++
+[source,yaml]
+----
+test_policy:
```
Comment (rosieyohannan) [inline jekyll/_cci2/test-config-policies.adoc:416]: @davidmdm Could you check what changes I need to make for this end to end test. Now I've based the examples on following on from the basic "write a policy" how to guide I think there are some small changes needed.

File: jekyll/_cci2/test-config-policies.adoc [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+CAUTION: While config policy management is in open preview stage, it should **not** be used for compliance purposes. The feature might become unavailable occasionally during **open preview**, during which build configurations may **not** be evaluated against policies.
+
+Follow the how-to guides on this page to write and set up tests for your config policies.
+
+This guide assumes you are familiar with writing policies for use with the config policy management feature. If you would like more information or information on configuration policy management in general, or the processes around writing policies, see the xref:config-policy-management-overview.adoc[Config policy management overview] and xref:use-the-cli-and-vcs-for-config-policy-management.adoc[Use the CLI and VCS for config policy management] pages.
+
+[#prerequisites]
+== Prerequisites
+
+The steps in this guide assume you have followed and completed the xref:/docs/use-the-cli-and-vcs-for-config-policy-management/#create-a-policy#[Create a policy] guide. In that guide you will have done the following:
```
Comment (rosieyohannan) [inline jekyll/_cci2/test-config-policies.adoc:25]: This testing how to guide should follow on nicely from this section: https://circleci.com/docs/use-the-cli-and-vcs-for-config-policy-management/#create-and-upload-a-policy

File: jekyll/_cci2/use-the-cli-and-vcs-for-config-policy-management.adoc [hunk 3]
```diff
@@ -15,29 +15,34 @@ NOTE: Config policy management is available on the **Scale** plan and is current
 
 CAUTION: While config policy management is in open preview stage, it should **not** be used for compliance purposes. The feature might become unavailable occasionally during **open preview**, during which build configurations may **not** be evaluated against policies.
 
-CAUTION: Ensure you have authenticated your version of the CLI with a token, and updated the CLI, before attempting to use the CLI for config policy management. See the link:/docs/local-cli[Installing the Local CLI] page for more information.
+Follow the how-to guides on this page to learn how to perform various config policy management processes.
 
-Use the CircleCI CLI to manage your organization's policies programmatically. The sub-commands to perform policy management are grouped under the `policy` command. 
+[#config-policy-management-enablement]
+== Enable or disable policy evaluation for an organization
```
Comment (rosieyohannan) [inline jekyll/_cci2/use-the-cli-and-vcs-for-config-policy-management.adoc:21]: Here I've just copied info on enabling the feature to the main feature page, from one of the how to guides

File: jekyll/_cci2/config-policy-management-overview.adoc [hunk 4]
```diff
@@ -38,13 +38,81 @@ Config policy management uses a decision engine based on link:https://www.openpo
...
++
+Example output:
++
+[source,shell]
+----
+{
+  "enabled": false
+}
+----
+* `test` - used to run tests on your policies.
```
Comment (rosieyohannan) [inline jekyll/_cci2/config-policy-management-overview.adoc:89]: Added test command here

File: jekyll/_cci2/config-policy-management-overview.adoc [hunk 5]
```diff
@@ -38,13 +38,81 @@ Config policy management uses a decision engine based on link:https://www.openpo
...
+reason:           string                                      (optional, only present when status is ERROR)
+enabled_rules:    Array<string>
+soft_failures:    Array<{ rule: string, reason: string }>
+hard_failures:    Array<{ rule: string, reason: string }>
+```
+
+The decision result tracks which rules were included in the making of the decision, and any violations, both soft and hard, that were found.
+
+[#use-the-cli-for-config-policy-management]
+== Use the CLI for config policy management
```
Comment (rosieyohannan) [inline jekyll/_cci2/config-policy-management-overview.adoc:54]: This section has been moved across from a how to guide

File: jekyll/_cci2/test-config-policies.adoc [hunk 6]
```diff
@@ -0,0 +1,613 @@
...
+input: null
+meta: null
+---- End of Test Context ---
+ok    config-policies    0.000s
+
+1/1 tests passed (0.000s)
+----
+
+== Add another policy and test
+Next, add a second policy and test to your `config-policies` directory. The steps below show hot to add a policy that specifies the minimum Docker version for xref:building-docker-images.adoc[remote Docker], writing tests for that policy, and running those tests.
```
Comment (davidmdm) [inline jekyll/_cci2/test-config-policies.adoc:171]: Detected a typo! how not hot

File: jekyll/_cci2/test-config-policies.adoc [hunk 7]
```diff
@@ -0,0 +1,607 @@
...
+[source,shell]
+----
+ok    config-policies            0.001s
+ok    config-policies/docker     0.001s
+ok    config-policies/version    0.001s
+
+11/11 tests passed (0.003s)
+----
+
+## Use metadata with tests
```
Comment (rosieyohannan) [inline jekyll/_cci2/test-config-policies.adoc:490]: @davidmdm I'm having trouble with this section, just getting lots of errors when running tests. Could you take a look?

**Review by davidmdm — APPROVED** [review]
LGTM! 🎉

**Comment by devinmarieb** [comment]
@rosieyohannan 
A few notes - I think it might be good to put "open preview" somewhere in the title of the page. We know readers often skip over banners. On Fail fast, we did "Fail fast - Open preview" and also put  "Fail fast (Open preview)" in the side nav.

This could just be the way I'm reading it, but I personally find the section titles "Develop configs" and "Develop policies" a bit confusing on the CLI page. Are users creating these things with the command line? I'm not really sure what "develop" means in this context I guess even after reading the section.

The Prerequisites section of the testing page has `[age]` which is a typo for “page” but also seems to be missing a link.

**Review by devinmarieb — APPROVED** [review]

**Pull request merged**